frontend/src/metabase/nav/components/SearchBar.tsx

```diff
@@ -194,16 +194,19 @@ function SearchBarView({ location, onSearchActive, onSearchInactive }: Props) {
           onKeyPress={handleInputKeyPress}
           ref={searchInput}
         />
-        <SearchFunnelButton
-          icon="filter"
-          data-is-filtered={isFiltered}
-          data-testid="search-bar-filter-button"
-          isFiltered={isFiltered}
-          onClick={e => {
-            e.stopPropagation();
-            setIsFilterModalOpen(true);
-          }}
-        />
+
+        {(!isSmallScreen() || isActive) && (
+          <SearchFunnelButton
+            icon="filter"
+            data-is-filtered={isFiltered}
+            data-testid="search-bar-filter-button"
```

Comment:
Maybe this should be an `aria-label` instead of `data-testid` since it would be useful to surface to screen readers, but I'm not sure if translations are frozen already.